src/status_im/protocol/handlers.cljs

```diff
@@ -179,19 +179,19 @@
 
 (re-frame/reg-fx
   ::participant-invited-to-group-message
-  (fn [{:keys [chat-id current-identity identity from message-id timestamp]}]
+  (fn [{:keys [group-id current-identity identity from message-id timestamp]}]
```

Comment:
Can you explain this change? I thought chat-id was used for "group id" as well. Can we check for presence of these in pre condition maybe?

Reply:
it comes from the event I changed below, group-id is in the payload not chat-id, maybe @janherich knows better I just fixed it because chat was empty and after fixing the initial bug I was getting the system messages such as "x added y" into a random chat (probably the default one for a nil public address)

Reply:
@oskarth for protocol communication, `:chat-id` is present in 1-1 chats and `:group-id` in group chats (with no `:chat-id` in payload).
I think that our data model in app-db (for chats) and protocol messages is unnecessary complicated and could be simpler and less error prone. 
It seems like it started by having `1-1` chat's and then "bolting" on additional data for group chats, without reconsidering it.
Imho just `:chat-id` everywhere with optional flag `:group?` would be totally sufficient.

Reply:
> :chat-id is present in 1-1 chats and :group-id in group chats payloads (with no :chat-id in payload).

Seems very error prone. Can we turn this into a pre-condition in this PR? As well as make a bounty to fix code base wide?

> Imho just :chat-id everywhere with optional flag :group?/:public? would be totally sufficient.
Agree, I think this could also be bounty. Suspect changing app db schema like this is a bit more work than just adding pre-conditions to catch this class of bug (hence suggesting two bounties, even though this would likely at least partially replace need for former).

Reply:
@oskarth I actually planned to do it while I was refactoring the message model (together with delivery status updates, so touching protocol as well), but the problem is that any changes in protocol will either make:

* Older app version incompatible with newer app version
* Introduce a lot of special handling in newer version which has to except protocol messages from older and newer app version alike (so checking for both `:chat-id` and `:group-id`...)

Given that bloated code and accidental complexity is enemy number 1, I decided against it, unless we drop the constraint of supporting everything we developed in the past (realm migrations are similar case) :)

Reply:
>  planned to do it
> problems

Guessing you refer to bounty 2 here, not 1. Yes agree, I think this is why it is long-term, we can figure out best upgrade path. But we can skip bounty 2 and only do bounty 1 if it isn't worth it.

> unless we drop the constraint of supporting everything we developed

We could consider this, but I think with a better upgrade store (realm read app start, better use of realm) then it should be specced pure function and errors at startup if we can't go from S0 -> S1.